pkg/standalone-utils/test/GaugeActions.test.ts

```diff
@@ -0,0 +1,659 @@
+  });
+
+  sharedBeforeEach('deploy token mocks', async () => {
+    vault = await Vault.create({ admin });
+
+    BAL = await deploy('v2-liquidity-mining/TestBalancerToken', {
+      args: [admin.address, 'Balancer', 'BAL'],
+    });
+
+    veBAL = await deploy('v2-liquidity-mining/TestBalancerToken', {
+      args: [admin.address, 'Balancer Voting Escrow', 'veBAL'],
+    });
+
+    rewardToken = await deploy('v2-liquidity-mining/TestBalancerToken', {
+      args: [admin.address, 'Reward', 'RWT'],
+    });
+
+    lpToken = await deploy('v2-pool-utils/MockBalancerPoolToken', {
+      args: ['Mock Balancer Pool Token', 'BPT', vault.address],
+    });
+
+    await lpToken.mint(userSender.address, totalLpTokens);
+  });
+
+  sharedBeforeEach('set up relayer', async () => {
+    adaptor = await deploy('v2-liquidity-mining/AuthorizerAdaptor', { args: [vault.address] });
+
+    gaugeController = await deploy('v2-liquidity-mining/MockGaugeController', {
+      args: [veBAL.address, adaptor.address],
+    });
+
+    const balTokenAdmin = await deploy('v2-liquidity-mining/MockBalancerTokenAdmin', {
+      args: [vault.address, BAL.address],
+    });
+    await BAL.connect(admin).grantRole(await BAL.MINTER_ROLE(), balTokenAdmin.address);
+
+    balMinter = await deploy('v2-liquidity-mining/BalancerMinter', {
+      args: [balTokenAdmin.address, gaugeController.address],
+    });
+
+    // Deploy Relayer: vault and BAL minter are required; we can skip wstETH.
+    relayerLibrary = await deploy('MockBatchRelayerLibrary', {
+      args: [vault.address, ZERO_ADDRESS, balMinter.address],
+    });
+    relayer = await deployedAt('BalancerRelayer', await relayerLibrary.getEntrypoint());
+
+    // Authorize Relayer for all actions
+    const relayerActionIds = await Promise.all(
+      ['setRelayerApproval', 'manageUserBalance'].map((action) => actionId(vault.instance, action))
+    );
+    await vault.grantPermissionsGlobally(relayerActionIds, relayer);
+
+    // Approve relayer by BPT holder
+    await vault.setRelayerApproval(userSender, relayer, true);
+  });
+
+  // We won't be needing the adder; we just need a staking liquidity gauge where tokens can be deposited / withdrawn.
+  sharedBeforeEach('set up gauge', async () => {
+    const veBalDelegation = await deploy('v2-liquidity-mining/MockVeDelegation');
+
+    const veBalDelegationProxy = await deploy('v2-liquidity-mining/VotingEscrowDelegationProxy', {
+      args: [vault.address, veBAL.address, veBalDelegation.address],
+    });
+
+    const gaugeImplementation = await deploy('v2-liquidity-mining/LiquidityGaugeV5', {
+      args: [balMinter.address, veBalDelegationProxy.address, adaptor.address],
+    });
+    gaugeFactory = await deploy('v2-liquidity-mining/LiquidityGaugeFactory', { args: [gaugeImplementation.address] });
+
+    gauge = await deployedAt(
+      'v2-liquidity-mining/LiquidityGaugeV5',
+      await deployGauge(gaugeFactory, lpToken.address) // No weight cap.
```

Comment:
You don't need to go through all this - you can just deploy the gauge directly, skipping the factory entirely.

Reply:
There is a small difference: gauges use proxies when created via factories, and the `create` method calls initialize with the right arguments. Otherwise we'd need to call it manually.

Even if it adds a few more lines, I'd vouch for leaving it like this for completion. WDYT?